lib/cli/placeorder.ts

```diff
@@ -1,20 +1,21 @@
 import { Arguments, Argv } from 'yargs';
 import { Order, OrderSide, PlaceOrderEvent, PlaceOrderRequest, PlaceOrderResponse, SwapFailure, SwapSuccess } from '../proto/xudrpc_pb';
 import { callback, loadXudClient } from './command';
-import { coinsToSats, satsToCoinsStr } from './utils';
+import { checkPrecision, coinsToSats, satsToCoinsStr } from './utils';
 
 export const placeOrderBuilder = (argv: Argv, side: OrderSide) => {
   const command = side === OrderSide.BUY ? 'buy' : 'sell';
   argv.option('quantity', {
-    type: 'number',
+    type: 'string',
```

Comment:
I think it's better to keep this as a number, since that's what it is, and make the string conversion (if necessary) later in the code.

Reply:
It's not that simple unfortunately, because if the user sets the quantity to 1e-24 for example it will be converted with .toString() to "1e-24".

Reply:
in case if we want to use toFixed(decimals), we will have to specify large number of decimals like 100 for potential user case when people will specify 10 ** 100 precision and check all digits after decimal point if they are not 0.

Reply:
You can do `toFixed` with 12 decimal places, then compare the price to the fixed string.

```typescript
const fixedPrice = price.toFixed(12);
if (price > fixedPrice) // we have numbers after the first 12 decimal places, invalid argument
```

Reply:
@sangaman I hope JS compares doubles correctly.

Reply:
What is your concern? All numbers are doubles in JS.

Reply:
@sangaman its better to avoid comparison of doubles with high precision